src/Report.php

```
@@ -0,0 +1,279 @@
+     *
+     * @var array
+     */
+    protected $policyChecks = [];
+
+  /**
+   * A flag indicating whether the policy checks were located or not.
+   *
+   * @var bool
+   */
+    protected $policyChecksLocated = false;
+
+    /**
+     * Report constructor.
+     *
+     * @param array $data An array containing data to be passed in for checks and reports.
+     */
+    public function __construct(array $data = [])
+    {
+        $this->data = $data;
+    }
+
+    /**
+     * Returns the data array.
+     *
+     * @return array The data.
+     */
+    public function getData()
+    {
+        return $this->data;
+    }
+
+    /**
+     * Sets a policy check.
+     *
+     * @param \EdisonLabs\PolicyVerification\Check\AbstractPolicyCheckBase $policyCheck The policy check object.
+     */
+    public function setCheck(AbstractPolicyCheckBase $policyCheck)
+    {
+        $this->policyChecks[] = $policyCheck;
+    }
+
+    /**
+     * Gets the policy check instances.
+     *
+     * @return array An array containing the policy check instances.
+     *
+     * @throws \Exception
+     */
+    public function getChecks()
+    {
+        if (!$this->policyChecksLocated) {
+            $containerBuilder = new ContainerBuilder($this->data);
+            $containerBuilder = $containerBuilder->getContainerBuilder();
+
+            $services = $containerBuilder->getServiceIds();
+
+            foreach ($services as $serviceName) {
+                // Filter by policy check classes only.
+                if (strpos($serviceName, self::POLICY_VERIFICATION_NAMESPACE) === false) {
+                    continue;
+                }
+
+                /** @var \EdisonLabs\PolicyVerification\Check\PolicyCheckInterface $policyCheck */
+                $policyCheck = $containerBuilder->get($serviceName);
+
+                // Sanity check by class type.
+                if (!$policyCheck instanceof AbstractPolicyCheckBase) {
+                    continue;
```

Can we throw an error here instead?